pkg/fqdn/name_manager.go

```diff
@@ -166,30 +165,15 @@ func (n *NameManager) GetDNSHistoryModel(endpointID string, prefixMatcher Prefix
 	return lookups, nil
 }
 
-// Lock must be held during any calls to RegisterForIPUpdatesLocked or
-// UnregisterForIPUpdatesLocked.
-// Because the NameManager and SelectorCache have interleaving locks, we
-// must ALWAYS lock the NameManager first before locking the SelectorCache.
-func (n *NameManager) Lock() {
-	n.RWMutex.Lock()
-}
-
-// Unlock must be called after calls to RegisterForIPUpdatesLocked or
-// UnregisterForIPUpdatesLocked are done.
-func (n *NameManager) Unlock() {
-	n.RWMutex.Unlock()
-}
+// RegisterFQDNSelector exposes this FQDNSelector so that the identity labels
+// of IPs contained in a DNS response that matches said selector can be
+// associated with that selector.
+// This function also evaluates if any DNS names in the cache are matched by
+// this new selector and updates the labels for those DNS names accordingly.
+func (n *NameManager) RegisterFQDNSelector(selector api.FQDNSelector) {
```

Comment:
We had discussed pre-allocating an identity for this selector. I think we should do this; it's a nice win and can help prevent policy drops on restart if IPs change identity.

Reply:
Oh yes, this fell between the cracks! Will add it to the PR

Reply:
Discussed offline, will add as a follow-up after this PR